tests/unit/test_handler_service_with_context.py

```diff
@@ -63,6 +63,20 @@ def test_test_initialize(inference_handler):
         inference_handler.initialize(CONTEXT)
         assert inference_handler.initialized is True
 
+@require_torch
+@pytest.mark.skipif(not _is_gpu_available(), reason="No GPU available")
+@slow
+def test_initialize_without_gpu_id_fallback_to_first_gpu(inference_handler):
```

Comment:
I never got to test this myself since I don't have GPU access currently. Also not sure if we want to use the `skipif` on availability of GPU. I didn't see any such marks in the test file. But i think it makes sense to have it